File: python/content_moderation_phrases.py
from soniox.transcribe_file import transcribe_file_short
from soniox.speech_service import SpeechClient


def main():
    with SpeechClient() as client:
        result = transcribe_file_short(
            "../test_data/test_audio.flac",
            client,
            content_moderation_phrases=["two years", "homesick"],
        )
        print("Words: " + " ".join(w.text for w in result.words))


if __name__ == "__main__":
    main()

File: node/content_moderation_phrases.js
const { SpeechClient } = require("@soniox/soniox-node");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
  const result = await speechClient.transcribeFileShort(
    "../test_data/test_audio.flac",
    {
      content_moderation_phrases: ["two years", "homesick"],
    }
  );

  console.log(`Words: ${result.words.map((word) => word.text).join(" ")}`);
})();
